docs/report-entries.adoc

```diff
@@ -0,0 +1,79 @@
+:page-title: Report entries
+:page-description: JUnit Jupiter extension to report things with annotations.
+
+You can use `@ReportEntry` as a simple way to declaratively add metadata to test methods.
+
+From the https://https://junit.org/junit5/docs/current/user-guide/#writing-tests-dependency-injection[JUnit 5 documentation]:
+
+"In JUnit Jupiter you should use `TestReporter` where you used to print information to
+`stdout` or `stderr` in JUnit 4. Using `@RunWith(JUnitPlatform.class)` will output all reported
+entries to `stdout`. In addition, some IDEs print report entries to `stdout` or display them
+in the user interface for test results."
+
+== Standard Use of TestReporter
+
+From the https://https://junit.org/junit5/docs/current/user-guide/#writing-tests-dependency-injection[JUnit 5 documentation]:
+
+"If a constructor or method parameter is of type `TestReporter`, the `TestReporterParameterResolver`
+will supply an instance of `TestReporter`. The `TestReporter` can be used to publish additional
+data about the current test run. The data can be consumed via the `reportingEntryPublished()`
+method in a `TestExecutionListener`, allowing it to be viewed in IDEs or included in reports."
+
+So, you would use it kind of like this:
+
+[source,java]
+----
+@Test
+void reportingTest(TestReporter reporter) {
+    reporter.publishEntry("Hello World!");
+    // YOUR TEST CODE HERE
+}
+----
+
+You can have a look at https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/TestReporter.html[the official documentation how TestReporter works].
+
+== With the extension
+
+It can be argued that publishing data about your tests should not be in your test code because it
+is simply not part of your test. It is meta-data.
+
+So it makes sense that you should be able to add that meta-data to your test declaratively with
+annotations and have JUnit take care of the publishing. This is what this extension is for!
+
+You can write...
+
+[source,java]
+----
+@Test
+@ReportEntry("Hello World!")
+void reportingTest() {
+    // YOUR TEST CODE HERE
+}
+----
+
+...and the extension will publish your report entry for you!
+
+You can declare multiple report entries on the same test (the annotation is repeatable).
+
+[source,java]
+----
+@Test
+@ReportEntry("foo")
+@ReportEntry("bar")
+void reportingTest() {
+    // YOUR TEST CODE HERE
+}
+----
+
+You can also declare your own keys in the annotation (if you don't declare it, JUnit
+will use the value of the entry as the key).
```

Comment:
Does ist really use the value of the entry as the key or does it use the string `value` as the key?

Reply:
You are right, the new solution does use the String "value" as a key if there is no key specified. My original solution used `publishReportEntry(String value)` <- that does behave this way.

I'll rewrite it.